pkg/envoy/cds/response.go

```diff
@@ -62,6 +62,15 @@ func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_d
 	}
 	clusters = append(clusters, localCluster)
 
+	if featureflags.IsRoutesV2Enabled() {
+		// if this service is associated with a trafficsplit service, add a local cluster for the apex service
+		for _, splitService := range meshCatalog.GetApexServicesForBackend(proxyServiceName) {
+			lcn := envoy.GetLocalClusterNameForService(splitService)
+			lc := getSyntheticCluster(lcn)
```

Comment:
I am not sure why a traffic split configuration should impact local clusters (on the uptream). Since traffic split is a client side configuration, why does this affect the configuration at the server side?

Reply:
When a request comes, we need an inbound virtual host programmed for the match of on the domains corresponding to the apex service in the traffic split on upstream and that match needs a route action that points to a local cluster. I did think about having the route match for the traffic split apex service domains point to the existing local cluster instead of a new one but decided to go this simpler route for now. I can take another look at going that route.

Reply:
That makes sense. However, what if the apex service is not just a virtual service but an actual service accepting traffic from clients - in that case programming a synthetic service seems misleading? There could be clients trying to reach the apex service without a traffic split, in which case this cluster should not be synthetic?

Reply:
The scenario you're describing I believe is multiple services per proxy which I think we should handle with this issue #2134. At the moment, I was configuring v1alpha2 in the routes refactor work as it would behave in the legacy code path so we could switch over asap to the routes v2 code path. Was planning to solve multiple services pointing to a single proxy in #2134. The e2e scenario here would not have overlapping labels configured for the apex service and backend services.

Reply:
Got it, thanks for clarifying. When we do implement #2134 we can tweak this.

Reply:
Can you add issue number 2134 to a comment for this code?